File: Archiver/LiteDB.vb

Module liteDB

    Public Class Zip_Store
        Dim RowNbr As Integer
        Dim DirID As Int32
        Dim FileID As Int32
        Dim FQN As String
        Dim EmailAttachment As Boolean
        Dim SuccessfullyProcessed As Boolean
        Dim fSize As Int64
        Dim CreateDate As DateTime
        Dim LastAccessDate As DateTime
        Dim NumberOfZipFiles As Integer
        Dim ParentGuid As String
        Dim InWork As Boolean
        Dim FileHash As String
    End Class


    Public Class DirFilesID_Store
        Dim FQN As String
        Dim LastArchiveDate As DateTime
        Dim FileLength As Int64
        Dim LastModDate As DateTime
    End Class

    Public Class Directory_Store
        Dim DirName As String
        Dim DirID As Integer
        Dim UseArchiveBit As Boolean
        Dim DirHash As String
        Dim LastArchiveDate As DateTime
    End Class

    Public Class Exts_Storage
        Dim Extension As String
        Dim Verified As Integer
    End Class

    Public Class FileNeedProcessing_Storage
        Dim RowID As Integer
        Dim FQN As String
        Dim LineID As Integer
        Dim LastProcessDate As DateTime
        Dim FileCompleted As Integer
    End Class

    Public Class Files_Storage
        Dim FileID As Integer
        Dim FileName As String
        Dim FileHash As String
    End Class

    Public Class Inventory_Storage
        Dim InvID As Integer
        Dim DirID As Integer
        Dim FileID As Integer
        Dim FileExist As Boolean
        Dim FileSize As Int64
        Dim CreateDate As DateTime
        Dim LastUpdate As DateTime
        Dim LastArchiveUpdate As DateTime
        Dim ArchiveBit As Boolean
        Dim NeedsArchive As Boolean
        Dim FileHash As String
    End Class

    Public Class ZipFile_Storage
        Dim RowNbr As Integer
        Dim DirID As Int32
        Dim FileID As Int32
        Dim FQN As String
        Dim EmailAttachment As Boolean
        Dim SuccessfullyProcessed As Boolean
        Dim fSize As Int64
        Dim CreateDate As DateTime
        Dim LastAccessDate As DateTime
        Dim NumberOfZipFiles As Integer
        Dim ParentGuid As String
        Dim InWork As Boolean
        Dim FileHash As String
    End Class

    Public Class Outlook_Store
        Dim RowID As Int32
        Dim sKey As String
        Dim KeyExists As Boolean
    End Class

    Public Class ContactsArchive_Store
        Dim RowID As Integer
        Dim Email1Address As String
        Dim FullName As String
    End Class

    '**********************************************

    Public Class DirListener_Store
        Dim RowID As Integer
        Dim ListenerFileName As String
        Dim LastIdProcessed As Integer
        Dim LastProcessDate As DateTime
        Dim FileCanBeDropped As Integer
        Dim RowCreateDate As DateTime
    End Class

    Public Class FileNeedProcessing_Store
        Dim RowID As Integer
        Dim ContainingFile As String
        Dim FQN As String
        Dim LineID As Integer
        Dim LastProcessDate As DateTime
        Dim FileApplied As Integer
        Dim RowCreateDate As DateTime
    End Class

    Public Class ProcessedListenerFiles_sTORE
        Dim RowID As Integer
        Dim ListenerFileName As String
        Dim RowCreateDate As DateTime
    End Class



End Module

File: Archiver/LiteDB.cs
using System;
using MODI;

namespace EcmArchiver
{
    static class liteDB
    {
        public class Zip_Store
        {
            private int RowNbr;
            private int DirID;
            private int FileID;
            private string FQN;
            private bool EmailAttachment;
            private bool SuccessfullyProcessed;
            private long fSize;
            private DateTime CreateDate;
            private DateTime LastAccessDate;
            private int NumberOfZipFiles;
            private string ParentGuid;
            private bool InWork;
            private string FileHash;
        }

        public class DirFilesID_Store
        {
            private string FQN;
            private DateTime LastArchiveDate;
            private long FileLength;
            private DateTime LastModDate;
        }

        public class Directory_Store
        {
            private string DirName;
            private int DirID;
            private bool UseArchiveBit;
            private string DirHash;
            private DateTime LastArchiveDate;
        }

        public class Exts_Storage
        {
            private string Extension;
            private int Verified;
        }

        public class FileNeedProcessing_Storage
        {
            private int RowID;
            private string FQN;
            private int LineID;
            private DateTime LastProcessDate;
            private int FileCompleted;
        }

        public class Files_Storage
        {
            private int FileID;
            private string FileName;
            private string FileHash;
        }

        public class Inventory_Storage
        {
            private int InvID;
            private int DirID;
            private int FileID;
            private bool FileExist;
            private long FileSize;
            private DateTime CreateDate;
            private DateTime LastUpdate;
            private DateTime LastArchiveUpdate;
            private bool ArchiveBit;
            private bool NeedsArchive;
            private string FileHash;
        }

        public class ZipFile_Storage
        {
            private int RowNbr;
            private int DirID;
            private int FileID;
            private string FQN;
            private bool EmailAttachment;
            private bool SuccessfullyProcessed;
            private long fSize;
            private DateTime CreateDate;
            private DateTime LastAccessDate;
            private int NumberOfZipFiles;
            private string ParentGuid;
            private bool InWork;
            private string FileHash;
        }

        public class Outlook_Store
        {
            private int RowID;
            private string sKey;
            private bool KeyExists;
        }

        public class ContactsArchive_Store
        {
            private int RowID;
            private string Email1Address;
            private string FullName;
        }

        // **********************************************

        public class DirListener_Store
        {
            private int RowID;
            private string ListenerFileName;
            private int LastIdProcessed;
            private DateTime LastProcessDate;
            private int FileCanBeDropped;
            private DateTime RowCreateDate;
        }

        public class FileNeedProcessing_Store
        {
            private int RowID;
            private string ContainingFile;
            private string FQN;
            private int LineID;
            private DateTime LastProcessDate;
            private int FileApplied;
            private DateTime RowCreateDate;
        }

        public class ProcessedListenerFiles_sTORE
        {
            private int RowID;
            private string ListenerFileName;
            private DateTime RowCreateDate;
        }
    }
}